internal/gatewayapi/testdata/gateway-with-two-listeners-with-tcproutes-with-sectionname.out.yaml

```diff
@@ -188,6 +188,10 @@ xdsIR:
             name: tcproute/default/tcproute-1/rule/-1/backend/0
             protocol: TCP
             weight: 1
+        metadata:
```

Comment:
its expensive to add this from a mem usage perspective, why this this needed

Reply:
I explained below why the metadata is needed.

Reply:
can we do it conditionally only if the feature is enabled? Maybe that also related to Arko's other comment about EnvoyProxy level feature as opposed to BTP.